### PR TITLE
Extend the grammer to explicitly support CURSOR variables in a declaration section.

### PR DESCRIPTION
**Pull request by hawicz**
To fix issue #45.

Also allow "COLLATE" and "DEFAULT" in regular variable declarations.

### CODE REVIEW

File: lib/piggly/parser/grammar.tt [hunk 1]
```diff
@@ -271,9 +271,21 @@ grammar Piggly
...
-               ( tSpace? kwASSIGN tSpace? rval:expressionUntilSemiColon )?
+    type:tType ( tSpace  kwCOLLATE name:tIdentifier )?
+               ( tSpace  kwNOT tSpace kwNULL )?
+               ( tSpace? ( kwASSIGN / kwDEFAULT ) tSpace? rval:expressionUntilSemiColon )?
+    tSpace?
+    ';' tSpace?
+  end
+
+  rule varDeclarationCursor
+    name:tIdentifier tSpace ( kwNO tSpace kwSCROLL tSpace )? kwCURSOR tSpace
```
Comment (kputnam) [inline lib/piggly/parser/grammar.tt:287]: According to [this document](https://www.postgresql.org/docs/current/plpgsql-cursors.html), `NO` is optional. So both `DECLARE foo SCROLL CURSOR ...` and `DECLARE foo NO SCROLL CURSOR ...` should be accepted. You should be able to change this to `( kwNO tSpace )?` I think.

Comment (hawicz) [reply lib/piggly/parser/grammar.tt:287]: Sure.  Will do.
Also, I need to make at least one other slight tweak, so please hold off on merging this just yet.

**Comment by hawicz** [comment]
> 
> 
> This looks good! Could you add a small test file to `example/proc/issue_45.sql` to test a number of permutations that should be accepted by the grammar? You can follow the steps in `example/README` to check that it works.

Ok, will do.

File: lib/piggly/parser/grammar.tt (hunk 1 above)
Comment (hawicz) [reply lib/piggly/parser/grammar.tt]: Ok, adjusted.  Please check out the current set of changes.

**Comment by kputnam** [comment]
Thanks Eric!

**Pull request merged**